test.sh

```diff
@@ -31,15 +31,27 @@ fi
 cd $(dirname $0)/v4
 DIR=$(mktemp -d -t softhXXXX)
 
-MODULE=${MODULE:-/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so}
+trap cleanUp EXIT
+function cleanUp {
+    if [ -d "${DIR}" ]; then
+        rm -rf ${DIR}
```

Comment:
Nice improvement to the cleanups!

Nit: the `-d` check is unnecessary since we know the directory will always exist; we created it above.

And `-f` is overkill. We expect to be able to remove the directory without `-f`, so providing `-f` doesn't provide a benefit and slightly increases the risk of doing something bad (descending into a write-protected directory).